test/serializer.cpp

```diff
@@ -427,6 +427,33 @@ class serializer_test
             BOOST_TEST(sr.done());
         }
 
+        {
+            value jv = std::numeric_limits<double>::quiet_NaN();
+            serializer sr;
+            sr.reset(&jv);
+            BOOST_TEST(! sr.done());
+            char buf[32];
+            BOOST_TEST(sr.read(buf) == "NaN");
+            BOOST_TEST(sr.done());
+        }
+        {
+            value jv = std::numeric_limits<double>::infinity();
+            serializer sr;
+            sr.reset(&jv);
+            BOOST_TEST(! sr.done());
+            char buf[32];
+            BOOST_TEST(sr.read(buf) == "Infinity");
+            BOOST_TEST(sr.done());
+        }
+        {
+            value jv = -std::numeric_limits<double>::infinity();
+            serializer sr;
+            sr.reset(&jv);
+            BOOST_TEST(! sr.done());
+            char buf[32];
+            BOOST_TEST(sr.read(buf) == "-Infinity");
```

Comment:
Is this what Javascript emits, `-Infinity` ?

Reply:
No, JS emits `null` for infinities and NaNs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#:~:text=The%20numbers%20Infinity%20and%20NaN%2C%20as%20well%20as%20the%20value%20null%2C%20are%20all%20considered%20null

I'm not actually sure whether these tests should be part of this PR though, considering they are not testing the parser, but decided to leave them just in case for now.